src/screens/FeedScreen/Header.js

```diff
@@ -191,8 +257,35 @@ const _renderProfileNormal = ({
   onHeaderOptionClicked = () => {},
   hideThreeDot,
   isPostDetail,
-  karmaScore = 0
+  karmaScore = 0,
+  isFollow = false,
+  onPressFollUnFoll = () => {},
+  onDeletePost = () => {},
+  isShowDelete = false,
+  isSelf = false
 }) => {
+  const refSheet = React.useRef();
+  const dataSheet = [
+    {
+      id: 1,
+      name: 'Share link',
+      icon: <ShareAndroidIcon />,
+      onPress: () => ShareUtils.shareUserLink(username)
+    }
+  ];
+
+  if (isShowDelete) {
+    dataSheet.push({
+      id: 2,
+      name: 'Delete post',
+      icon: <TrashRed />,
+      onPress: () => {
+        refSheet.current.close();
+        onDeletePost();
+      },
+      style: {color: colors.red}
+    });
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similar to the previous comment, avoid mutating the `dataSheet` array directly. Use a more functional approach.
```diff
-  if (isShowDelete) {
-    dataSheet.push({
-      id: 2,
-      name: 'Delete post',
-      icon: <TrashRed />,
-      onPress: () => {
-        refSheet.current.close();
-        onDeletePost();
-      },
-      style: {color: colors.red}
-    });
-  }
+  const deleteOption = {
+    id: 2,
+    name: 'Delete post',
+    icon: <TrashRed />,
+    onPress: () => {
+      refSheet.current.close();
+      onDeletePost();
+    },
+    style: {color: colors.red}
+  };
+
+  const finalDataSheet = isShowDelete ? [...dataSheet, deleteOption] : dataSheet;
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->